luaui/Widgets/unit_factory_assist_fix.lua

```diff
@@ -0,0 +1,121 @@
+local widget = widget ---@type Widget
+
+function widget:GetInfo()
+	return {
+		name      = "Factory Assist Fix",
+		desc      = "Fixes factory assist so that builders don't leave to repair damaged finished units",
+		author    = "TheDujin",
+		date      = "Jun 30 2025",
+		license   = "GNU GPL, v2 or later",
+		layer     = 0,
+		enabled   = true
+	}
+end
+
+----------------------------------------------------------------
+-- Globals
+----------------------------------------------------------------
+local myTeam = Spring.GetMyTeamID()
+
+-- Tracks unit IDs of all assist-capable builders that I own and are alive
+local myAssistBuilders = {}
+
+local isAssistBuilder = {}
+
+----------------------------------------------------------------
+-- Speedups
+----------------------------------------------------------------
+local spGetUnitCurrentCommand = Spring.GetUnitCurrentCommand
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+local spAreTeamsAllied = Spring.AreTeamsAllied
+local spGetUnitHealth = Spring.GetUnitHealth
+local spGetUnitDefID = Spring.GetUnitDefID
+local spGetUnitTeam = Spring.GetUnitTeam
+
+local CMD_REPAIR = CMD.REPAIR
+local CMD_GUARD = CMD.GUARD
+local CMD_REMOVE = CMD.REMOVE
+local CMD_OPT_ALT = CMD.OPT_ALT
+
+for unitDefID, unitDef in pairs(UnitDefs) do
+	if unitDef.isBuilder and unitDef.canAssist then
+		isAssistBuilder[unitDefID] = true
+	end
+end
+
+-- If this builder unit is repairing the newly built unit when it should
+-- instead be guarding the factory, remove the repair command
+local function maybeRemoveRepairCmd(builderUnitID, builtUnitID, factID)
+	local firstCmdID, _, _, firstCmdParam_1 = spGetUnitCurrentCommand(builderUnitID, 1)
+   if firstCmdID ~= CMD_REPAIR or firstCmdParam_1 ~= builtUnitID then
```

Comment:
Perhaps looking if the repair command is internal would help vs the case where manually-given repair is followed by guard?
```suggestion
	local firstCmdID, firstCmdOpts, _, firstCmdParam_1 = spGetUnitCurrentCommand(builderUnitID, 1)
   if firstCmdID ~= CMD_REPAIR or firstCmdParam_1 ~= builtUnitID or math.bit_and(firstCmdOpts, CMD.OPT_INTERNAL) ~= 0 then
```

Reply:
What does "Internal Order" mean? If that flag is set to true, does that mean the command came from a non-player source (e.g. a widget)? Or does that mean the command came from Engine? Or what does it mean?

Reply:
Engine commands that spawn sub-commands (like Guard spawns Repair) is supposed to add the internal flag to the sub-command while anything else generally isn't. If it doesn't work that's alright though, probably no big deal.